tests/cases/compiler/noUsedBeforeDefinedErrorInTypeContext.ts

```diff
@@ -0,0 +1,23 @@
+// @noEmit: true
+
+// https://github.com/microsoft/TypeScript/issues/8775
+
+interface IThing<T> {
+    owner: T;
+}
+
+var foo = {
+    one: {} as IThing<typeof foo>,
+}
+
+let baz = {
+    two: {} as IThing<typeof bar>,
+}
+
+let bar = {
+    three: {} as IThing<typeof bar>,
+}
```

Comment:
this isn't actually super useful as the result of this is an implicit any but that's a separate error category and the reported "no used before defined" was still redundant

Reply:
can you set `@strict: true` to show that the no implicit any errors are there?